src/CollapsibleItem.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Item from './Item';
+import { PropertyName } from './properties';
+
+class CollapsibleItem extends Component {
+  constructor(props) {
+    super(props);
+    // component can be controlled or uncontrolled depends on did we pass collapsed state on init or not
+    // similar to react input components
+    this.controlled = typeof this.props.collapsed !== 'undefined';
+    this.state = {
+      collapsed: this.props.collapsed
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
It would mean support only for the newest react as I understand. And I'm not sure we can update bblfsh-dashboard easily. I'm 100% in to move to `getDerivedStateFromProps` but we need to make it right. Let's do it in separate ticket later.

Reply:
That makes sense.